kep/108-Memory-bandwidth-and-LLC-aware-filtering-and-scoring/README.md

```diff
@@ -0,0 +1,160 @@
+# Memory bandwidth and LLC aware filtering and scoring
+
+## Table of Contents
+
+<!-- toc -->
+- [RDT aware filtering and scoring](#RDT-aware-filtering-and-scoring)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+  - [Terms](#terms)
+  - [Proposal](#proposal)
+  - [Design Details](#design-details)
+    - [Extension points](#extension-points)
+      - [Filter](#filter)
+      - [PreScore](#prescore)
+      - [Score](#score)
+  - [Known Limitations](#known-limitations)
+  - [Alternatives considered](#alternatives-considered)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Testing Plan](#testing-plan)
+  - [Implementation History](#implementation-history)
+  - [References](#references)
+<!-- /toc -->
+
+## Summary
+
+This document describes filtering and scoring base on memory bandwidth and LLC (last level cache).
+
+## Motivation
+
+Memory bandwidth and LLC are very important resources in a system. Kuberenetes scheduler hasn't taken them into consideration yet. But memory bandwidth and LLC contention may greatly impact system performance. Moveover, it may cause poor SLA. For instance, in a CI/CD usage scenario, run 1 pipeline the totoal build time is 7min, run 16 pipelines the build time increases to 21min. Each pipeline (pod) is assigned the same CPU and memory, but the performance is quite different. We find once memory bandwidth is used out the performance will drop significantly, the more pipelines the worse performance. Intel® Resource Director Technology (Intel® RDT https://www.intel.com/content/www/us/en/architecture-and-technology/resource-director-technology.html) is one example of the technology that can be leveraged to mitigate the impact of memory bandwidth and LLC contention. It is very helpful to ensure system performance stability and improve SLA. 
+
+## Goals
+1. Use scheduler plugin, which is the most Kubernetes native way, to implement memory bandwidth and LLC aware filtering and scoring.
+2. Leverage memory bandwidth and LLC relative metrics for filtering and scoring.
+3. Provide configurable weights for prioritizing the metrics used in the scoring calculations.
+
+## Non-Goals
+
+## Use Cases
+1. If a workload is memory bandwidth bond, e.g. rpmbuild, it uses much memory bandwidth. If too much such kind of workloads are scheduled to a node, the memory bandwith of the node will be used out quickly. The performance of the workloads will drop significantly. If the scheduler uses this filter plugin it can be avoid. 
+
+2. If there are many workloads run on a node, the LLC contention may become very intense. The performance of all workloads will drop. If the scheduler uses this score plugin, it can ensure new workload is scheduled to LLC freer node. Furtherly, mitigate performance drop.
```

Comment:
IIUC, this may be reworded to:

```
2. If workloads that are affinitive to particular resources, such as LLC contention, are scheduled to the same node, the performance of those workloads may drop significantly. We'd prefer to schedule workloads with similar resource affinity to different nodes to mitigate performance downgrade.
```

Reply:
done. Thanks!